### DISCUSSION TITLE
Persistent "libcrypto loading in an unsafe way" Error with Ruby on Rails on macOS

### DISCUSSION THREAD

**Discussion by sparsh9** [rbenv/rbenv#1519] [Q&A] [answered]
Hello,

I am using a 2017 MacBook Pro with an Intel chip, running Ruby version 2.7.8, Rails 5.2.6, and OpenSSL 1.1. I've installed Ruby using rbenv and ruby-build.

I am persistently receiving the following error when trying to create a database with Rails:

```
WARNING: /Users/logan/.rbenv/versions/2.7.8/bin/ruby is loading libcrypto in an unsafe way
zsh: abort bundle exec bin/rails db:create
```

Steps I have taken to troubleshoot this issue: 
1. Installed OpenSSL via Homebrew and ensured that the version is 1.1. 
2. Uninstalled and reinstalled Ruby with rbenv, specifying the correct OpenSSL directory. 
3. Set the DYLD_LIBRARY_PATH  environment variable to the path where Homebrew installed the OpenSSL libraries before running the Ruby process.

**Comment by mislav** [comment, marked as answer]
Hello, I do not understand this warning message, but it sounds like this is a runtime problem rather than rbenv problem.

[This reddit user says](https://www.reddit.com/r/rubyonrails/comments/15ct7m8/persistent_libcrypto_loading_in_an_unsafe_way/) “upgrade your eth gem”— I don't know what this gem is, but you could potentially try this? Good luck!

**Reply by sparsh9** [reply]
yeah, it got resolved. @mislav
